manifests/server/config.pp

```diff
@@ -1,10 +1,9 @@
 class puppet::server::config {
 
   include puppet
```

Comment:
do your tests make sure that `include puppet` is happening beofre puppet::server::config is processed?
btw, we can ensure this happens by marking this class with [assert_private()](https://github.com/puppetlabs/puppetlabs-stdlib#assert_private)